### PR TITLE
add test case for user object

### PR DESCRIPTION
**Pull request by DaveXiong**

### CODE REVIEW

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java [hunk 1]
```diff
@@ -13,6 +13,8 @@
 	private String password;
 
 	private Collection<Role> roles;
+	
+	private int a;
```
Comment (DaveXiong) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: MAJOR Remove this unused "a" private field. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by DaveXiong** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java (hunk 1 above)
Comment (DaveXiong) [reply sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: ACK

Comment (DaveXiong) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: MAJOR Remove this unused "a" private field. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by DaveXiong** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java (hunk 1 above)
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: MAJOR Remove this unused "a" private field. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by dxiongtech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java (hunk 1 above)
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: MAJOR Remove this unused "a" private field. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by dxiongtech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java (hunk 1 above)
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: MAJOR Remove this unused "a" private field. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by dxiongtech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by dxiongtech** [comment]
test

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java (hunk 1 above)
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/entity/User.java]: MAJOR Remove this unused "a" private field. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1068)

**Comment by dxiongtech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java [hunk 2]
```diff
@@ -27,6 +28,14 @@ public User create(String email, String password) {
 		user.setPassword(password);
 
 		email2User.put(email, user);
+
+		while(true) {
```
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java]: BLOCKER Add an end condition to this loop. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS2189)

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java [hunk 3]
```diff
@@ -27,6 +28,14 @@ public User create(String email, String password) {
...
+			//do it
+		}
+		if (tmp.equals("tmp")) {
```
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java]: MAJOR Move the "tmp" string literal on the left side of this string comparison. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1132)

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java [hunk 4]
```diff
@@ -27,6 +28,14 @@ public User create(String email, String password) {
...
+			System.out.println("tmp is tmp");
```
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java]: MAJOR Replace this use of System.out or System.err by a logger. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS106)

**Comment by dxiongtech** [comment]
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java [hunk 5]
```diff
@@ -18,6 +18,7 @@
 	private Map<String, User> email2User = new HashMap<String, User>();
 
 	public User create(String email, String password) {
+		String tmp;
```
Comment (dxiongtech) [inline sonarqube-sample-java/src/main/java/com/consoleconnect/mw/sonarqube/sample/service/UserService.java:21]: MAJOR Remove this unused "tmp" local variable. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1481)

**Comment by dxiongtech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**